ui/app/serializers/sync/destination.js

```diff
@@ -12,17 +12,14 @@ export default class SyncDestinationSerializer extends ApplicationSerializer {
   };
 
   serialize(snapshot) {
-    // special serialization only for PATCH requests
-    if (snapshot.isNew) return super.serialize(snapshot);
+    const data = super.serialize(snapshot);
 
-    // only send changed values
-    const data = {};
-    for (const attr in snapshot.changedAttributes()) {
-      // first array element is the old value
-      const [, newValue] = snapshot.changedAttributes()[attr];
-      data[decamelize(attr)] = newValue;
-    }
-    return data;
+    if (snapshot.isNew) return data;
+    // only send changed parameters for PATCH requests
+    const patchData = {};
+    const changedKeys = Object.keys(snapshot.changedAttributes()).map((key) => decamelize(key));
+    changedKeys.forEach((key) => (patchData[key] = data[key]));
```

Comment:
I think it's probably fine either way, but this could be nice as a reducer function:
```suggestion
    const patchData = changedKeys.reduce((payload, key) => {
    payload[key] = data[key];
    return payload;
    }, {})
```

Reply:
Sure, I can update this. I always think reduce functions are hard to follow so I typically avoid them 🙈

Reply:
They just take a little getting used to 😁 I started to love them once it clicked